examples/eady_turbulence.jl

```diff
@@ -5,306 +5,398 @@
-                                       buoyancy = BuoyancyTracer(),
-                                        tracers = :b,
-                                        forcing = (u=Fu_eady, v=Fv_eady, w=Fw_eady, b=Fb_eady),
-                                        closure = closure,
-                            boundary_conditions = (u=ubcs, v=vbcs, b=bbcs))
+using Oceananigans.Advection: WENO5
+
+model = IncompressibleModel(
+           architecture = CPU(),
+                   grid = grid,
+              advection = WENO5(),
+            timestepper = :RungeKutta3,
+               coriolis = coriolis,
+                tracers = :b,
+               buoyancy = BuoyancyTracer(),
+      background_fields = (b=B_field, u=U_field),
+                closure = (Laplacian_vertical_diffusivity, biharmonic_horizontal_diffusivity),
+    boundary_conditions = (u=u_bcs, v=v_bcs)
+)
 
 # # Initial conditions
 #
-# For initial conditions we impose a linear stratifificaiton with some
-# random noise.
+# We use non-trivial initial conditions consisting of an array of vortices superposed
+# with large-amplitude noise to (hopefully) stimulate the rapid growth of
+# baroclinic instability.
 
-## A noise function, damped at the boundaries
-Ξ(z) = rand() * z/Lz * (z/Lz + 1)
+## A noise function, damped at the top and bottom
+Ξ(z) = randn() * z/grid.Lz * (z/grid.Lz + 1)
 
-## Buoyancy: linear stratification plus noise
-b₀(x, y, z) = N² * z + 1e-2 * Ξ(z) * (N² * Lz + α * f * Lh)
-u₀(x, y, z) = 1e-2 * α * Lz
+## Scales for the initial velocity and buoyancy
+Ũ = 1e-1 * background_parameters.α * grid.Lz
+B̃ = 1e-2 * background_parameters.α * coriolis.f
 
-set!(model, u=u₀, v=u₀, b=b₀)
+uᵢ(x, y, z) = Ũ * Ξ(z)
+vᵢ(x, y, z) = Ũ * Ξ(z)
+bᵢ(x, y, z) = B̃ * Ξ(z)
 
-# # Diagnostics and output
-#
-# Diagnostics that return the maximum absolute value of `u, v, w` by calling
-# `umax(model), vmax(model), wmax(model)`:
+set!(model, u=uᵢ, v=vᵢ, b=bᵢ)
+
+# We subtract off any residual mean velocity to avoid exciting domain-scale
+# inertial oscillations. We use a `sum` over the entire `parent` arrays or data
+# to ensure this operation is efficient on the GPU (set `architecture = GPU()`
+# in `IncompressibleModel` constructor to run this problem on the GPU if one
+# is available).
 
-umax = FieldMaximum(abs, model.velocities.u)
-vmax = FieldMaximum(abs, model.velocities.v)
-wmax = FieldMaximum(abs, model.velocities.w)
+Ū = sum(model.velocities.u.data.parent) / (grid.Nx * grid.Ny * grid.Nz)
+V̄ = sum(model.velocities.v.data.parent) / (grid.Nx * grid.Ny * grid.Nz)
 
-# Set up output. Here we output the velocity and buoyancy fields at intervals of one day.
-output_writer = JLD2OutputWriter(model, merge(model.velocities, model.tracers);
-                                 time_interval=day, prefix=output_filename_prefix,
-                                 force=true, max_filesize=10GiB)
+model.velocities.u.data.parent .-= Ū
+model.velocities.v.data.parent .-= V̄
```

Comment:
Should we be encouraging the use of `.data.parent` in an example? Would it be better to set up the array on the CPU and use `set!(::Field, ::Array)`? This would be messy too though...

Reply:
The issue isn't GPU versus CPU; the issue is that we initialize with a random field but still need to exactly zero out whatever mean momentum might remain. This is because even small amounts of mean momentum excite inertial oscillations.. which are inert, but corrupt the visualizations.

For this small case I think we could write a convenience function called, perhaps, `zero_out_mean`. I decided not to include in this PR because I thought it'd be good to discuss...

The more general problem is to provide an interface for analytical transformations of data. We could, perhaps, provide a `set!` method that uses `AbstractOperations`. Then maybe this code could look like

```julia
set!(model, u = (x, y, z) -> randn())

volume_mean_U = mean(u, dims=(1, 2, 3))

zero_mean_u = u - volume_mean_U

set!(model, u = zero_mean_u)
```

Definitely open to other ideas as well...